awx/main/dispatch/worker/base.py

```diff
@@ -152,9 +157,18 @@ def run(self, *args, **kwargs):
                         self.process_task(json.loads(e.payload))
                     if self.should_stop:
                         return
+                    self.pg_is_down = False
             except psycopg2.InterfaceError:
                 logger.warn("Stale Postgres message bus connection, reconnecting")
                 continue
+            except (db.DatabaseError, psycopg2.OperationalError):
+                if not self.pg_is_down:
+                    logger.exception("Postgres error processing tasks")
+                    self.pg_down_time = time.time()
+                    self.pg_is_down = True
+                if time.time() - self.pg_down_time > 60.0:
```

Comment:
We should make the 60 configurable, and maybe we share that constant with `update_model`. But I'm worried about doing any settings references in that cancel_callback (due to threading issues).